src/libraries/System.Threading.Channels/src/System/Threading/Channels/UnboundedChannel.cs

```diff
@@ -112,24 +114,60 @@ public override ValueTask<T> ReadAsync(CancellationToken cancellationToken)
 
             public override bool TryRead([MaybeNullWhen(false)] out T item)
             {
-                UnboundedChannel<T> parent = _parent;
+                UnboundedChannel<T, TQueue> parent = _parent;
+                return parent._items.IsThreadSafe ?
+                    IsThreadSafe(parent, out item) :
+                    NotIsThreadSafe(parent, out item);
```

Comment:
again, another subjective rename proposal

```suggestion
                    ThreadSafe(parent, out item) :
                    NotThreadSafe(parent, out item);
```

some other ideas: `Locked` and `LockFree`

Reply:
Renamed. Thanks.